step-automation-packages/step-automation-packages-controller/src/main/java/step/automation/packages/AutomationPackageExecutionPlugin.java

```diff
@@ -30,40 +33,70 @@ public AutomationPackageExecutionPlugin(AutomationPackageLocks automationPackage
     @Override
     public void executionStart(ExecutionContext context) {
         super.executionStart(context);
-        String automationPackageId = getAutomationPackageId(context);
-        if (automationPackageId != null) {
+        if (shouldLock(context)) {
+            String automationPackageId = getAutomationPackageId(context);
             try {
+                debugLog(context, "Trying to acquire read lock on automation package.");
                 boolean locked = automationPackageLocks.tryReadLock(automationPackageId);
                 if (!locked) {
-                    logger.debug("Timeout while acquiring read lock on automation package " + automationPackageId);
+                    debugLog(context, "Timeout while acquiring read lock on automation package.");
                     throw new PluginCriticalException("Timeout while acquiring lock on automation package with id " +
                             automationPackageId + ". This usually means that an update of this automation package is on-going and took more than the property " +
                             AUTOMATION_PACKAGE_READ_LOCK_TIMEOUT_SECS + " (default " + AUTOMATION_PACKAGE_READ_LOCK_TIMEOUT_SECS_DEFAULT + " seconds)");
                 }
-                logger.debug("Acquired read lock on automation package " + automationPackageId);
+                context.put(EXECUTION_CONTEXT_LOCK_ID, automationPackageId);
```

Comment:
it doesn't change anything in the code flow here as we throw an exception in the if-clause but I would generally recommend to put this in the else block